### DISCUSSION TITLE
Mongodb instrospection, relationship with same name of idfield and relation field

### DISCUSSION THREAD

**Discussion by masterbater** [prisma/prisma#12745] [Q&A]
Questions or request for instrospection
 - Is it possible in the future that prisma will have instrospect using mongoose schema or at least add docs to migrate from mongoose  schema to prisma schema

How to map relationship if my relation field is client instead of clientId?

With mongoose we define a reference or a relationship like this, with autogenerated timestamps for createdAt and updatedAt

```ts
const SetupFeeSchema = new mongoose.Schema(
  {
    price: Number,
    client: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Client"
    }
  },
  { timestamps: true }
);
```

Schema generated using npx prisma db pull
```prisma
model clients {
  id                    String   @id @default(auto()) @map("_id") @db.ObjectId
  v                     Int      @map("__v")
  username              String   @unique(map: "username_1")
  createdAt DateTime @db.Date
  updatedAt DateTime @db.Date
  organization String
}

model setupfees {
  id        String   @id @default(auto()) @map("_id") @db.ObjectId
  v         Int      @map("__v")
  client    String   @db.ObjectId
  createdAt DateTime @db.Date
  price     Float
  updatedAt DateTime @db.Date
}
```

With this to add a relationship using prisma way

```prisma
model setupfees {
  id        String   @id @default(auto()) @map("_id") @db.ObjectId
  v         Int      @map("__v")
  clientId String @db.ObjectId
  client   clients @relation(fields: [userId], references: [id])
  createdAt DateTime @db.Date
  price     Float
  updatedAt DateTime @db.Date
}
```

Does this mean I need to change my idFields to with client to clientId in my db  . Please let me know if I could make it something like this duplicate field client in schema where it defines objectId then the reference
```prisma
model setupfees {
  id        String   @id @default(auto()) @map("_id") @db.ObjectId
  v         Int      @map("__v")
  client String @db.ObjectId
  client   clients @relation(fields: [client], references: [id])
  createdAt DateTime @db.Date
  price     Float
  updatedAt DateTime @db.Date
}
```

**Comment by janpio** [comment]
Hey, interesting case. You want to use the name of the field that hold the data, as the name of the relation field for Prisma as well.

One way is to do it like this:
```prisma
model setupfees {
  id        String   @id @default(auto()) @map("_id") @db.ObjectId
  v         Int      @map("__v")
  clientId  String   @db.ObjectId @map("client")
  client    clients  @relation(fields: [clientId], references: [id])
  createdAt DateTime @db.Date
  price     Float
  updatedAt DateTime @db.Date
}
```

Note the `@map("client")` that I added to the `String   @db.ObjectId` field that used to be called just `client` but now is `clientId` - this is what tells Prisma to internally call the field `clientId` but in the database store the data in `client` so it works without changes to your data. Also note that I adapted the `fields` value of the relation to use the correct "relation scalar field" that is now called `clientId`.

Let me know if this works for you please!

**Reply by masterbater** [reply]
Thanks a lot, I appreciate so much your help.

**Comment by janpio** [comment]
> Is it possible in the future that prisma will have instrospect using mongoose schema 

That is a _great_ idea. Maybe not as a built in functionality for the Prisma CLI, but as a separate tool that takes your mongoose schema and can then turn it into a Prisma schema. Can you please open a feature request issue about this here? If you can include an example for a mongoose schema that would be awesome!

> or at least add docs to migrate from mongoose schema to prisma schema

Yes, that work is tracked here: https://github.com/prisma/docs/issues/2793

**Comment by YoungMahesh** [comment]
Hi guys, do you know what is the meaning of `@map("__v")` which is auto-generated by prisma ? Is it necessary to keep or we can remove it ?

**Reply by janpio** [reply]
You have a field `__v`, which Prisma can not express. So it renames it to `v` but tells the database that the field should actually be called `__v` there.

**Reply by YoungMahesh** [reply]
But in my example or in the above example shared by @masterbater:
```js
const SetupFeeSchema = new mongoose.Schema(
  {
    price: Number,
    client: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Client"
    }
  },
  { timestamps: true }
);
```

you can see there is no `__v` field in the mongoose Schema, still it generated `  v    Int      @map("__v")` in prisma-schema

**Reply by janpio** [reply]
This comes from https://mongoosejs.com/docs/guide.html#versionKey which I assume is active by default in Mongoose or in your configuration.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
